x/mongo/driver/topology/topology.go

```diff
@@ -91,6 +91,18 @@ type Topology struct {
 var _ driver.Deployment = &Topology{}
 var _ driver.Subscriber = &Topology{}
 
+type serverSelectionConfig struct {
+	selector    description.ServerSelector
+	timeoutChan <-chan time.Time
+}
```

Comment:
I think this was a good change. In general I favor grouping state together in structs like this. My only nit here might be that `State` is a better suffix than `Config` since it contains a channel, but I don't care too much, personally.

Reply:
Changed to `serverSelectionState` and updated all references to be called `selectionState` rather than `cfg`.